### PR TITLE
[statistics] In the stats panel, fix hard-coded expression return value as numerical

### PR DESCRIPTION
**Pull request by nirvn**
## Description

This PR fixes the bogus assumption/limitation of the statistics panel which treated all expressions as having a numerical return type.

### CODE REVIEW

File: src/app/qgsstatisticalsummarydockwidget.cpp [hunk 1]
```diff
@@ -159,6 +159,46 @@ void QgsStatisticalSummaryDockWidget::refreshStatistics()
   {
     mFieldType = fieldType( mFieldExpressionWidget->currentField() );
   }
+  else
+  {
+    QgsExpressionContext context;
```
Comment (nyalldawson) [inline src/app/qgsstatisticalsummarydockwidget.cpp:164]: Would be nice to move this logic out to QgsExpressionUtils and avoid the duplicate code. Something like "determineResultType"

File: src/core/qgsaggregatecalculator.cpp [hunk 2]
```diff
@@ -158,7 +139,7 @@ QVariant QgsAggregateCalculator::calculate( QgsAggregateCalculator::Aggregate ag
           case StringConcatenateUnique:
           case StringMinimumLength:
           case StringMaximumLength:
-            v = QString();
+            v = "";
```
Comment (nyalldawson) [inline src/core/qgsaggregatecalculator.cpp:142]: is this change required?

Comment (nirvn) [reply src/core/qgsaggregatecalculator.cpp:142]: No, indeed not.

Comment (nirvn) [reply src/core/qgsaggregatecalculator.cpp]: @nyalldawson , fixed.

**Review by nyalldawson — APPROVED** [review]
